packages/next/client/image.tsx

```diff
@@ -89,13 +89,29 @@ allSizes.sort((a, b) => a - b)
 
 function getWidths(
   width: number | undefined,
-  layout: LayoutValue
+  layout: LayoutValue,
+  sizes: string | undefined
 ): { widths: number[]; kind: 'w' | 'x' } {
   if (
     typeof width !== 'number' ||
     layout === 'fill' ||
     layout === 'responsive'
   ) {
+    if (sizes && layout === 'fill') {
```

Comment:
What about `layout === 'responsive'?

Maybe this if statement should be moved to the first line of `getWidths()`

Reply:
This should be addressed now. Let me know if the new version wasn't what you were thinking of